scripts/ci/check-regression-promotions.py

```
@@ -0,0 +1,215 @@
+import sys
+from typing import Any
+
+
+def require_text(value: Any, field: str, errors: list[str]) -> str:
+    text = str(value or "").strip()
+    if not text:
+        errors.append(f"missing {field}")
+    return text
+
+
+def validate(manifest_path: pathlib.Path, today: dt.date) -> list[str]:
```

Ruff (0.16.0)

[warning] Too many branches (43 > 12)

(PLR0912)